src/sentry/plugins/providers/integration_repository.py

```diff
@@ -133,18 +135,25 @@ def create_repository(
             try:
                 with transaction.atomic(router.db_for_write(Repository)):
                     repo = Repository.objects.create(
-                        organization_id=organization.id, name=result["name"], **repo_update_params
+                        organization_id=organization.id, **repo_update_params
                     )
             except IntegrityError:
                 # Try to delete webhook we just created
                 try:
-                    repo = Repository(
-                        organization_id=organization.id, name=result["name"], **repo_update_params
-                    )
+                    repo = Repository(organization_id=organization.id, **repo_update_params)
                     self.on_delete_repository(repo)
                 except IntegrationError:
                     pass
 
+                # if possible update the repo with matching integration
+                repo = Repository.objects.filter(
```

Comment:
@cathteng I think you can just use `.filter(...).update(...)`

Reply:
oh so i update all of them

Reply:
@cathteng your uniqueness constraint means there would only be 0 or 1 matches, just cleaner code